src/Uno.UI/Helpers/WeakEvents/WeakEventManager.cs

```diff
@@ -6,130 +6,110 @@
 using System.Reflection;
 using System.Runtime.CompilerServices;
 
-// Mostly from https://github.com/dotnet/maui/blob/c92d44c68fe81c57ef8caec2506e6788309b8ff4/src/Core/src/WeakEventManager.cs
-// But adjusted a little bit
+namespace Uno.UI.Helpers;
 
-namespace Uno.UI.Helpers
+internal sealed class WeakEventManager
 {
-	internal sealed class WeakEventManager
-	{
-		private readonly Dictionary<string, List<Subscription>> _eventHandlers = new(StringComparer.Ordinal);
-
-		public void AddEventHandler(Action? handler, [CallerMemberName] string eventName = "")
-		{
-			if (string.IsNullOrEmpty(eventName))
-				throw new ArgumentNullException(nameof(eventName));
+	private readonly ConditionalWeakTable<object, Dictionary<string, List<Action>>> _handlersPerTarget = new();
+	private Dictionary<string, List<Action>>? _staticHandlers;
+	private string? _enumeratingHandlersOfEvent;
 
-			if (handler == null)
-				throw new ArgumentNullException(nameof(handler));
+	public void AddEventHandler(Action handler, [CallerMemberName] string eventName = "")
```

Comment:
Let's keep it nullable. The problem is that events are usually declared nullable, so the compiler thinks `value` in the context of add/remove is also nullable. Making this nullable helps not having to suppress nullability warning in callsites.

Reply:
Getting `null` shouldn't happen, so throwing is fine I think.

It's mostly an artifact that the `value` parameter is having the same type as the event. The nullability in the event type is mostly around when there are no subscribers so that `MyEvent.Invoke` doesn't crash. But whenever we do `+= Something` or `-= Something`, the `Something` is the value and should never be null, even if the type is nullable.